opentracing/src/main/java/opentracing/Span.java

```diff
@@ -62,29 +60,12 @@
   Span setTraceAttribute(String key, String value);
 
   /**
-   * {@code message} is a format string and can refer to fields in the payload by path, like so:
-   *
-   * <pre>{@code
-   *
-   * span.info("first transaction is worth ${transactions[0].amount} ${transactions[0].currency}",
-   *     ImmutableMap.of(
-   *       "transactions", asList(
-   *         Transaction.builder().amount(10).currency("USD").build(),
-   *         Transaction.builder().amount(11).currency("USD").build(),
-   *       )
-   *     )
-   * );
-   * }</pre>
-   *
-   * @param message {@link Formatter format string} that can refer to fields in the args payload.
-   * @param args arbitrary payload
-   */
-  // See https://github.com/opentracing/opentracing.github.io/issues/30 about parameterization
-  Span info(String message, Object... args);
-
-  /** Same as {@link #info}, but for warnings. */
-  Span warning(String message, Object... args);
+   * Add a timestamped annotation, with an optional payload.
+   **/
+  Span log(String message, /* @Nullable */ Object payload);
```

Comment:
This is what I had in mind, thinking forward past https://github.com/opentracing/opentracing.github.io/issues/30#issuecomment-188464752:

```
Span logEvent(String eventName, /* *Nullable */ Object payload);
Span info(String message, Object... params);
Span error(String message, Object... params);
// maybe / maybe-not warn(), etc... TBD.
Span log(
  /* @Nullable */ String eventName,
  /* @Nullable */ String formattedMessage,  // *not* a pre-formatted message+params pair
  bool error,  // could also be a Level if we go for more than info+error... or a @Nullable Bool
  /* @Nullable */ Object payload,
  /* @Nullable */ Instant timestamp)
```

i.e., there is an utterly unwieldy plumbing method (as you would put it) that handles the union of everything else, then specialized logging methods for the common cases of event and debug message logging.